src/yo/adapter.ts

```diff
@@ -44,38 +44,45 @@ export default class CodeAdapter {
 	public prompt(questions, callback) {
 		let answers = {};
 
-		var promise = questions.reduce((promise, question) => {
-			return promise.then(() => {
-				if (question.when === undefined) {
-					return true;
-				} else if (isFn(question.when)) {
-					return runAsync(question.when)(answers);
-				}
-
-				return question.when;
-			}).then(askQuestion => {
-				if (askQuestion) {
-					const prompt = PromptFactory.createPrompt(question);
-
-					return prompt.render().then(result => answers[question.name] = question.filter ? question.filter(result) : result);
-				}
-			});
-		}, Promise.resolve());
-
-		promise
-			.then(() => {
-				this.outChannel.clear();
-				this.outChannel.append(this.outBuffer);
-
-				callback(answers);
-			})
-			.catch(err => {
-				if (err instanceof EscapeException) {
-					return;
-				}
+		return new Promise((resolve, reject) => {
```

Comment:
Add `callback = callback || () => {};` after `let answers = {};`. This way, you don't have to check if `callback` exists every time.